bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/handler/RoombaHandler.java

```
@@ -503,4 +529,49 @@ private void reportProperty(String property, @Nullable String value) {
             updateProperty(property, value);
         }
     }
+
+    @Override
+    public void connectionStateChanged(MqttConnectionState state, @Nullable Throwable error) {
+        if (state == MqttConnectionState.CONNECTED) {
+            MqttBrokerConnection connection = this.connection;
+
+            if (connection == null) {
+                // This would be very strange, but Eclipse forces us to do the check
+                logger.error("Established connection without broker pointer");
+                return;
```

This sould be warn. You could also throw an unchecked exception like `IllegalStateException` if it's strange this would happen.

---

Done